packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -355,15 +355,26 @@ contract ERC20Vault is BaseVault {
             balanceChange_ = _op.amount;
         } else {
             // If it's a canonical token
-            IERC20Metadata meta = IERC20Metadata(_op.token);
             ctoken_ = CanonicalERC20({
                 chainId: uint64(block.chainid),
                 addr: _op.token,
-                decimals: meta.decimals(),
-                symbol: meta.symbol(),
-                name: meta.name()
+                decimals: 0,
+                symbol: "",
+                name: ""
             });
 
+            // Try fill in the boilerplate values, but use try-catch because functions below are
+            // ERC20-optional only.
+            try IERC20Metadata(_op.token).decimals() returns (uint8 _decimals) {
+                ctoken_.decimals = _decimals;
+            } catch { }
+            try IERC20Metadata(_op.token).name() returns (string memory _name) {
+                ctoken_.name = _name;
```

Comment:
how about we take only the first 32 bytes in case the returned name is longer...